Azael/app.js

```diff
@@ -0,0 +1,199 @@
+    counter[letters[i]] = (counter[letters[i]] || 0) + 1;
+  }
+
+  return counter;
+};
+console.log("countLetters :>> ", countLetters("tree"));
+
+// I've cheated....:)
+
+// **5. Free Shipping.**
+// Create a function that determines whether an online order should get free shipping. An order gets free shipping if the total cost of items exceeds €50.
+
+// Examples:
+
+// - freeShipping({ "Sponge": 3.50, "Soap": 5.99 }) ➞ false
+// - freeShipping({ "Surround Sound Equipment": 499.99 }) ➞ true
+// - freeShipping({ "Wool": 13.99, "Knitting Needles": 15.50, "Bag": 13.99 }) ➞ false
+
+function freeShipping(orderObj) {
+  let getMoney = Object.values(orderObj);
+  let addMoney = 0;
+  for (let i = 0; i < getMoney.length; i++) {
+    addMoney += parseFloat(getMoney[i]);
+  }
+  if (addMoney >= 50) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+console.log("freeShipping :>> ", freeShipping({ Sponge: 3.5, Soap: 5.99 }));
+console.log(
+  "freeShipping :>> ",
+  freeShipping({ "Surround Sound Equipment": 499.99 })
+);
+
+// **6. Programming Object.**
+
+// ```javascript
+const programming = {
+  languages: ["JavaScript", "Python", "Ruby"],
+  isChallenging: true,
+  isRewarding: true,
+  difficulty: 8,
+  jokes:
+    "http://stackoverflow.com/questions/234075/what-is-your-best-programmer-joke",
+};
+// ```
+
+// - Write the command to add the language "Go" to the end of the languages array.
+
+programming.languages[3] = "Go";
+console.log("programming :>> ", programming);
+
+// - Change the difficulty to the value of 7.
+programming.difficulty = 8;
+console.log("programming :>> ", programming);
+
+// - Using the delete keyword, write the command to remove the jokes key from the programming object.
+
+delete programming.jokes;
+console.log("programming :>> ", programming);
+// - Write a command to add a new key called isFun and a value of true to the programming object.
+
+programming.isFun = true;
+console.log("programming :>> ", programming);
+
+// - Using a loop, iterate through the languages array and console.log all of the languages.
+
+let languages = Object.values(programming.languages);
+
+console.log("languages :>> ", languages);
+
+// - Using a loop, console.log all of the keys in the programming object.
+
+let keysProgramming = Object.keys(programming);
+console.log("keysProgramming :>> ", keysProgramming);
+
+// - Using a loop, console.log all of the values in the programming object.
+
+let valuesProgramming = Object.values(programming);
+console.log("valuesProgramming :>> ", valuesProgramming);
+
+// - Create an object method where if the keys "isChallenging" and "isRewarding" have values of "true", then return "Learning the programming languages: "JavaScript, Python, Ruby, Go" is rewarding and challenging. **Bonus:** In a comment, explain what is printed if we console.log an object method without calling it and why.
+
+// **Bonus**:
+// - Make sure that any other code cannot delete or change properties of the object.
```

Comment:
The last Bonus Task==> 
Object.seal(programming);
delete programming.languages;
console.log(programming)